azurelinuxagent/common/utils/restutil.py

```diff
@@ -216,18 +216,31 @@ def http_request(method,
 
     msg = ''
     attempt = 0
-    delay = retry_delay
+    delay = 0
+    was_throttled = False
 
     while attempt < max_retry:
         if attempt > 0:
-            logger.info("[HTTP Retry] Attempt {0} of {1}: {2}",
+            # Compute the request delay
```

Comment:
This is the second change to this code relatively quickly.  What happened with the previous assumption?

Reply:
Sadly. The first change reduced the delay when the server throttled the request. But, in reviewing the code, I became suspicious that it might not delay long enough in edge cases. These changes bring the *total* delay times (over all attempts) back in line with the previous code while leaving the throttle induced delay short.